packages/namespaces-react/src/withNamespaceProvider.js

```diff
@@ -0,0 +1,19 @@
+import { mergeLeft } from 'ramda';
+import React from 'react';
+
+import NamespaceProvider from './NamespaceProvider';
+
+const withNamespaceProvider = optionProps => WrappedComponent => ({
+	feature,
+	namespace,
+	store,
+	useNamespace,
+	children,
+	...otherProps
+}) => (
```

Comment:
you need to do the usual HOC wrapping logic here, i.e. `const WithNamespaceProvider = props => ...`, assigning `displayName`. see other decorators in here or in our client project.